editor/scene_tree_dock.cpp

```diff
@@ -742,6 +742,14 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 				if (existing != empty) {
 					undo_redo->add_do_method(n, "set_script", empty);
 					undo_redo->add_undo_method(n, "set_script", existing);
+
+					List<PropertyInfo> properties;
+					n->get_property_list(&properties);
+					for (const PropertyInfo &property : properties) {
+						if (property.usage & (PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR)) {
```

Comment:
```suggestion
						if (property.usage & PROPERTY_USAGE_SCRIPT_VARIABLE && property.usage & (PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR)) {
```
This filters only for script variables.

Also is this intended that you check for either of STORAGE or EDITOR? If yes, you can use PROPERTY_USAGE_DEFAULT, which is equivalent.

Reply:
Oh, great, somehow it didn't cross my mind that we have a separate flag for script variables. I'll add this.

I guess `PROPERTY_USAGE_DEFAULT` would do. There was another place in the same file which used `PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR` so I thought I'd do it the same way. IMO it's more readable, you don't need to think what flags are in `PROPERTY_USAGE_DEFAULT`.

Reply:
It seems that `PROPERTY_USAGE_SCRIPT_VARIABLE` is set only during runtime, not in editor. So it's no use here.